spy.go

```diff
@@ -1,19 +1,78 @@
 package spy
 
 import (
+	"bufio"
 	"fmt"
 	"io"
+	"os"
 	"os/exec"
+	"os/user"
+	"time"
 
 	"bitbucket.org/creachadair/shell"
+	"github.com/fatih/color"
 )
 
-// ParseCommand accepts a string and returns the command name and args
+// sorry no American English, not today >.<
+type Colour color.Attribute
+
+const (
+	Cyan   Colour = 46
+	Yellow Colour = 43
+	Red    Colour = 101
+	Green  Colour = 42
+	White  Colour = 107
+)
+
+type Option func(*session) error
+
+type session struct {
+	TimestampMode bool
+	ShellPrompt   string
+	Colour        Colour
+	Input         io.Reader
+	Recorder      io.Writer
+	Terminal      io.Writer
+}
+
+func WithTimestamps() Option {
+	return func(s *session) error {
+		s.TimestampMode = true
+		return nil
+	}
+}
+
+func WithUserPrompt(userPrompt string) Option {
+	return func(s *session) error {
+		s.ShellPrompt = userPrompt
+		return nil
+	}
+}
+
+func WithTerminalColour(userColour Colour) Option {
+	return func(s *session) error {
+		switch userColour {
```

Comment:
Maybe we don't even need a `switch` here. You could just let the user select whatever colour they want, regardless of whether it's on the list or not. Or if you want to restrict them to just these values, you could create a map (of `[Colour]bool`) and validate that the selected colour is in the map.